test/infrastructure/docker/main.go

```diff
@@ -34,9 +34,11 @@ import (
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
 	expv1 "sigs.k8s.io/cluster-api/exp/api/v1alpha4"
 	"sigs.k8s.io/cluster-api/feature"
-	infrav1 "sigs.k8s.io/cluster-api/test/infrastructure/docker/api/v1alpha4"
+	infrav1alpha3 "sigs.k8s.io/cluster-api/test/infrastructure/docker/api/v1alpha3"
+	infrav1alpha4 "sigs.k8s.io/cluster-api/test/infrastructure/docker/api/v1alpha4"
```

Comment:
Note. currently there are no conversion for CAPD, but I don't see harm in being ready for implementing conversions, so I'm ok with this change

Reply:
noted. do you mind if I create a new issue for adding a conversion webhook for CAPD? consider the goal of CAPD is to be a reference implementation of an infrastructure provider, it would be great if CAPD has a conversion webhook as well.

Reply:
+100 from my side. thanks for offering to open an issue for tracking this

Reply:
created issue here https://github.com/kubernetes-sigs/cluster-api/issues/4436